service/src/main/java/bio/terra/profile/app/controller/AzureApiController.java

```diff
@@ -18,24 +21,38 @@ public class AzureApiController implements AzureApi {
   private final HttpServletRequest request;
   private final AzureService azureService;
   private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  private final PolicyServiceConfiguration policyServiceConfiguration;
 
   @Autowired
   public AzureApiController(
       HttpServletRequest request,
       AzureService azureService,
-      AuthenticatedUserRequestFactory authenticatedUserRequestFactory) {
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      PolicyServiceConfiguration policyServiceConfiguration) {
     this.request = request;
     this.azureService = azureService;
     this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+    this.policyServiceConfiguration = policyServiceConfiguration;
   }
 
   @Override
   public ResponseEntity<AzureManagedAppsResponseModel> getManagedAppDeployments(
       UUID azureSubscriptionId, Boolean includeAssignedApplications) {
     final AuthenticatedUserRequest userRequest = authenticatedUserRequestFactory.from(request);
-    var managedApps =
-        this.azureService.getAuthorizedManagedAppDeployments(
-            azureSubscriptionId, includeAssignedApplications, userRequest);
+    List<AzureManagedAppModel> managedApps;
+
+    // check azure control plane feature flag
+    if (this.policyServiceConfiguration.getAzureControlPlaneEnabled()) {
```

Comment:
it does not seem like `PolicyServiceConfiguration` is the right place for the config setting `azureControlPlaneEnabled`

Reply:
@marctalbott @sergiygetlin any suggestions on where to put the config for the AzureControlPlane flag?

Reply:
ExternalConfiguration?

Reply:
`AzureConfiguration` would likely be the best existing location for it. Or you could set up a new configuration class like `AzureControlPlaneConfiguration` specifically for control plane-related config. Either is fine with me

Reply:
Should we update AzureApiController test as well since original flow splits into two parts here (line 45) or per Marc's suggestion maybe we should move it to service level and this is where this split is controlled via specific mocking in the test - `when(policyServiceConfiguration.getAzureControlPlaneEnabled()).thenReturn(true|false)`.